src/dex-ui/pages/dao/DAOsListPage/__tests__/DAOsListPage.test.tsx

```diff
@@ -0,0 +1,42 @@
+import { render, screen, fireEvent, cleanup, renderHook } from "@testing-library/react";
+import { DAOsListPage } from "../DAOsListPage";
+import { vi } from "vitest";
+
+vi.mock("react-router-dom", async () => {
+  const actual: any = await vi.importActual("react-router-dom");
+  return {
+    ...actual,
+    useNavigate: () => vi.fn(),
+    useLocation: () => vi.fn(),
+  };
+});
+
+vi.mock("@hooks", async () => {
```

Comment:
In the future, when we create real tests, the mocking should occur when we call external services (MirrorNode service, Smart Contracts, etc.) to fetch data. For example, tests for the DAOsListPage would mock the data received from the [MirrorNode](https://testnet.mirrornode.hedera.com/api/v1/docs/#/contracts/listContractLogs) `/api/v1/contracts/${accountId}/results/logs` endpoint:

```
// This function would be mocked to return mocked MirrorNodeEventLog type data
const logs: MirrorNodeEventLog[] = await fetchNextBatch(`/api/v1/contracts/${accountId}/results/logs`, "logs", {
      params: {
        order: "desc",
      },
    }); 
```

We want our tests to cover the true functionality of hooks, child components, and other functions within the pages without interacting with services outside of our application. This will ensure that we are covering the majority of critical code paths from the perspective of the users' most valued features.

CC: @roshan-bisht-lab49

Reply:
Note: In a future iteration, we should strictly limit the purpose of the functions in service files, such as the  `MirrorNodeService`, to calling and returning data from external services - removing additional logic such as parsing, filtering, sorting, etc. This will simplify writing tests for components that utilize these services since we can assume all functions in these service files need to be mocked. This does not apply to the `DexService` since the `DexService` is a helper service that wraps the other service files and contains additional data transformation logic that we want to indirectly cover with RTL tests.